### ISSUE TITLE
False positive "not all trait items implemented" with defaulted assoc. const

### ISSUE THREAD

**Issue by SludgePhD** [rust-lang/rust-analyzer#15945]
**rust-analyzer version**: 0.3.1740-standalone

**rustc version**: rustc 1.74.0 (79e9716c9 2023-11-13)

```rust
trait Assoc {
    const ASSOC: u32 = 0;
}

impl Assoc for () {}
// not all trait items implemented, missing: `const ASSOC`
```

**Comment by lnicola** [comment]
Duplicate of #15909, a fix is available in the pre-release version.
